### PR TITLE
#167165005 Super admin can update and delete user account

### PR DESCRIPTION
**Pull request by OvieMudi**
#### What does this PR do?
- adds super admin endpoints and operations for updating and deleting
#### Description of Task to be completed?
- In additon, to the create and read operation routes, ensure the following endpoints are working

`PATCH  /api/v1/users/:userId` 
`DELETE  /api/v1/users/:userId` 

#### How should this be manually tested?
To test manually on your local computer:
-  run the following commands.

```
# Clone this repository
$ git clone https://github.com/andela/dahlia-ah-backend.git

# Navigate to the application folder
$ cd dahlia-ah-backend

# Checkout the branch
$ git checkout feature/167165005/super-admin-update-and-delete

# Install dependencies
$ npm install

# Start the app in development mode
$ npm run dev:start
```
- check the `.env.example` file on the environmental variables to provide
- on Postman make an authorized request to the endpoints above

Note: See the screenshot below on how the  request/response should look
#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#167165005 ](https://www.pivotaltracker.com/story/show/167165005 )

#### Screenshots
Screenshot 2019-08-16 at 12 35 46 AM

#### API Flowchart

zzzzAPI Flowchart

### CODE REVIEW

**Comment by Ukhu** [comment]
Kindly add a screenshot to the PR description of your flowchart for this task

File: src/middlewares/userAcess.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import services from '../services';
+import helpers from '../helpers';
+
+const { userServices: { findUserRole } } = services;
+const { errorResponse } = helpers;
+
+const verifySuperAdmin = async (req, res, next) => {
```
Comment (benisi) [inline src/middlewares/userAcess.js]: i think we already have a middleware called authorizeUser that can do this same function

**Review by Rythae — COMMENTED** [review]
Kindly include swagger documentation for this PR

File: src/middlewares/userAcess.js (hunk 1 above)
Comment (OvieMudi) [reply src/middlewares/userAcess.js]: Yeah, will rebase

**Comment by OvieMudi** [comment]
> Kindly include swagger documentation for this PR

Thank you very much for this observation. Right now, I'm waiting for more PR to be merged before adding the docs.
If you've observed, in this PR, I have re-routed the endpoints for signup and login and signup. They are now under the `auth` routes.
As you know, changes of this nature require already existing docs to be edited and moved about.
Also, my task involves docs for the other half of my task, i.e creating and getting users.

To summarize:
1. The docs for this feature is ready
2. It is not yet added because features are being merged chronologically. This will cause huge conflicts if I add the docs now.

I hope I was able to make you understand why I left out the docs.
Please let me know if you still feel I should add them now so that we can discuss more of the merits and demerits.
Thanks again for the observation.

**Comment by Rythae** [comment]
> > Kindly include swagger documentation for this PR
> 
> Thank you very much for this observation. Right now, I'm waiting for more PR to be merged before adding the docs.
> If you've observed, in this PR, I have re-routed the endpoints for signup and login and signup. They are now under the `auth` routes.
> As you know, changes of this nature require already existing docs to be edited and moved about.
> Also, my task involves docs for the other half of my task, i.e creating and getting users.
> 
> To summarize:
> 
> 1. The docs for this feature is ready
> 2. It is not yet added because features are being merged chronologically. This will cause huge conflicts if I add the docs now.
> 
> I hope I was able to make you understand why I left out the docs.
> Please let me know if you still feel I should add them now so that we can discuss more of the merits and demerits.
> Thanks again for the observation.

Alright. Well done.

**Comment by Nkemjiks** [comment]
Resolve merge conflicts..

**Comment by Nkemjiks** [comment]
You still have some merge conflicts..

**Review by WilliamsOhworuka — APPROVED** [review]

**Comment by Nkemjiks** [comment]
@OvieMudi Fix merge conflict
